### ISSUE TITLE
Meta-data property change callback called with empty key when another JACK client closes

### ISSUE THREAD

**Issue by SpotlightKid** [jackaudio/jack2#504]
Can be tested with [this script](https://gist.github.com/SpotlightKid/dc70dac006337c99bf65bcdc98e43ddc) for the [jackclient-python](https://github.com/spatialaudio/jackclient-python) bindings and then running, for example, `jack_property -l` or `jack_lsp` in another shell:

```console
$ python metadata-listener.py 
Press enter to quit...
Property '' on subject 8589934598 deleted.
```

You an also use `jackclient-python` as the JACK client to trigger this:

```pycon
>>> import jack
>>> client = jackClient('mytest')
>>> client.close()
```

~It only a happens the first time the client is activated, if you call `client.deactivate()` and then `client.activate()` again, the property change callback is not triggered, not is it triggered when the client closes.~

The `jackclient-python` bindings use `cffi` to call the jacklib functions and set up the callback function, so I'm sure a client written in C would behave the same. I also experience this with the [jacklib Python bindings](https://github.com/SpotlightKid/jack-matchmaker/blob/master/jackmatchmaker/jacklib.py) using `ctypes` used and included in `jack-matchmaker`

**Comment by SpotlightKid** [comment]
Hang on, testing again now this triggered when closing the client. I'll try to nail this down...

**Comment by SpotlightKid** [comment]
I'm not sure what I was seeing first, but now I can only trigger this, when a client closes.

**Comment by falkTX** [comment]
note that you can also use `jack_evmon` which since a few commits ago supports properties too

**Comment by SpotlightKid** [comment]
Yes, that also demonstrates this:

```console
$ jack_evmon
Graph reordered
Client mytest registered
Graph reordered              # <-- client activated
Graph reordered              # <-- client deactivated
Graph reordered              # <-- client closed
key [] for 8589934614 deleted
Client mytest unregistered
Graph reordered
```

**Comment by falkTX** [comment]
Fixed in 294d0ffd96091ca17eb6c5dcda312b1740796ebe
